src/back/wgsl/writer.rs

```diff
@@ -324,25 +324,30 @@ impl<W: Write> Writer<W> {
                 Attribute::Binding(id) => format!("binding({})", id),
                 Attribute::Group(id) => format!("group({})", id),
                 Attribute::Interpolate(interpolation, sampling) => {
-                    if interpolation.is_some() || sampling.is_some() {
-                        let interpolation_str = if let Some(interpolation) = interpolation {
-                            interpolation_str(interpolation)
-                        } else {
-                            ""
-                        };
-                        let sampling_str = if let Some(sampling) = sampling {
-                            // Center sampling is the default
-                            if sampling == crate::Sampling::Center {
-                                String::from("")
-                            } else {
-                                format!(",{}", sampling_str(sampling))
-                            }
+                    let sampling_part = match sampling {
+                        Some(sampling) if sampling != crate::Sampling::default() => {
+                            Some(sampling_str(sampling))
+                        }
+                        _ => None,
+                    };
+                    let parts: Vec<_> = vec![
```

Comment:
can we refactor this to avoid heap allocation? I understand that this particular case may not be too problematic, but generally across Naga codebase we try to avoid heap allocation in frontends and backends, unless necessary